src/lib/util.ts

```diff
@@ -32,4 +32,24 @@ export default class Util {
         return "Unknown network"
     }
   }
+
+  /**
+   * Subscribes to pending transactions for a specific operation.
+   * Unsubscribes automatically.
+   * @param topic Arc.js TransactionService topic
+   * @param key A unique `key` to look for in `txEventInfo.options`
+   * @param cb A callback to execute on each pending transaction.
+   */
+  public static onPendingTransactions(topic: string, key: symbol, cb: (txCount: number) => any) {
+    let count = 0;
+    const sub = Arc.TransactionService.subscribe(topic, (topic: string, txEventInfo: any) => {
+      if (txEventInfo.options.key === key && txEventInfo.tx) {
+        cb(txEventInfo.txCount);
```

Comment:
Could be good for the user to supply `count` to `cb` here in addition to the total count.  Then when you handle the event, in the message you can tell the user how far along they are (like "completed 2nd of 8" transactions)

Reply:
That's already done automatically. Each `dispatch` of a pending operation increments the `step` until the operation terminates with either `success` or `failure`.